ios/browser/url_sanitizer/url_sanitizer_service_factory.mm

```diff
@@ -0,0 +1,83 @@
+
+#include "base/no_destructor.h"
+#include "brave/components/url_sanitizer/browser/url_sanitizer_component_installer.h"
+#include "brave/components/url_sanitizer/browser/url_sanitizer_service.h"
+#include "brave/ios/browser/api/url_sanitizer/url_sanitizer_service+private.h"
+#include "brave/ios/browser/application_context/brave_application_context_impl.h"
+#include "brave/ios/browser/keyed_service/keyed_service_factory_wrapper+private.h"
+#include "brave/ios/browser/url_sanitizer/url_sanitizer_service_factory.h"
+#include "components/keyed_service/core/keyed_service.h"
+#include "components/keyed_service/ios/browser_state_dependency_manager.h"
+#include "ios/chrome/browser/shared/model/application_context/application_context.h"
+#include "ios/chrome/browser/shared/model/browser_state/browser_state_otr_helper.h"
+#include "ios/chrome/browser/shared/model/browser_state/chrome_browser_state.h"
+#include "ios/chrome/browser/shared/model/browser_state/chrome_browser_state_manager.h"
+#include "ios/web/public/browser_state.h"
+
+#if !defined(__has_feature) || !__has_feature(objc_arc)
+#error "This file requires ARC support."
+#endif
+
+@implementation URLSanitizerServiceFactory
++ (nullable id)serviceForBrowserState:(ChromeBrowserState*)browserState {
+  // Create and start the local data file service and component installer
+  brave::URLSanitizerService* urlSanitizer =
+      brave::URLSanitizerServiceFactory::GetServiceForState(browserState);
+  return [[URLSanitizerService alloc] initWithURLSanitizerService:urlSanitizer];
+}
+@end
+
+namespace brave {
+
+// static
+brave::URLSanitizerService* URLSanitizerServiceFactory::GetServiceForState(
+    ChromeBrowserState* browser_state) {
+  return static_cast<brave::URLSanitizerService*>(
+      GetInstance()->GetServiceForBrowserState(browser_state, true));
+}
+
+// static
+URLSanitizerServiceFactory* URLSanitizerServiceFactory::GetInstance() {
+  static base::NoDestructor<URLSanitizerServiceFactory> instance;
+  return instance.get();
+}
+
+URLSanitizerServiceFactory::URLSanitizerServiceFactory()
+    : BrowserStateKeyedServiceFactory(
+          "URLSanitizerService",
+          BrowserStateDependencyManager::GetInstance()) {}
+
+URLSanitizerServiceFactory::~URLSanitizerServiceFactory() = default;
+
+std::unique_ptr<KeyedService>
+URLSanitizerServiceFactory::BuildServiceInstanceFor(
+    web::BrowserState* context) const {
+  std::unique_ptr<brave::URLSanitizerService> service =
+      std::make_unique<brave::URLSanitizerService>();
+  BraveApplicationContextImpl* braveContext =
+      static_cast<BraveApplicationContextImpl*>(GetApplicationContext());
+  braveContext->url_sanitizer_component_installer()->AddObserver(service.get());
+  return service;
+}
+
+bool URLSanitizerServiceFactory::ServiceIsNULLWhileTesting() const {
+  return true;
+}
+
+bool URLSanitizerServiceFactory::ServiceIsCreatedWithBrowserState() const {
+  // Return true here because the service is instantiated lazily
```

Comment:
this comment is backwards, you return true to eagerly instantiate it, but the comment here is not necessary. The return value is self explanatory

Reply:
Removed!